src/mongo/db/query/explain_options.cpp

```diff
@@ -80,8 +81,23 @@ StatusWith<ExplainOptions::Verbosity> ExplainOptions::parseCmdBSON(const BSONObj
                                         << kAllPlansExecutionVerbosityStr
                                         << "'}");
         }
+        verbositySpecified = true;
     }
 
+    int numFields = 0;
+    BSONObjIterator i(cmdObj);
+    while (i.moreWithEOO()) {
+        BSONElement e = i.next();
+        if (e.eoo())
+            break;
+        if (e.fieldNameStringData() == "$db")
```

Comment:
The "lsid" field would also have to be allowed. There are some tests in jstests/core that are titled explain that may be helpful to run to show what needs to work.